tutorials/kotlin-springboot-container-engine.md

```diff
@@ -0,0 +1,383 @@
+
+        kubectl expose deployment demo --type=LoadBalancer --port 80 --target-port 8080
+
+    This creates a service resource pointing at your running pod. It listens
+    on the standard HTTP port 80, and proxies back to your pod on port 8080.
+
+1.  Obtain the IP address of the service by running:
+
+        kubectl get service
+
+    Initially, the external IP field will be pending while Kubernetes Engine
+    procures an IP address for you. If you rerun the `kubectl get service`
+    command repeatedly, eventually the IP address will appear. You can then
+    point your browser at that URL to view the running application.
+
+Congratulations! Your application is now up and running!
+
+## Scaling and updating your application
+
+You'll now explore a few of the basic features of Kubernetes for managing your
+running app.
+
+### Set the replica count
+
+Initially your deployment runs a single instance of your application. You can
+add more replicas using the `kubectl scale` command. For example, to add two
+additional replicas (for a total of three), run:
+
+    kubectl scale deployment demo --replicas=3
+
+Once the additional replicas are running, you can see the list of three pods
+by running:
+
+    kubectl get pods
+
+Kubernetes automatically allocates your running pods on the virtual machines
+in your cluster. You can configure pods in your deployment with specific
+resource requirements such as memory and CPU. See the
+[Kubernetes documentation](https://kubernetes.io/docs/home/) for more details.
+
+### Update your application
+
+After you make a change to your app, redeploying is just a matter of
+building a new image and pointing your deployment to it.
+
+1.  Make a change to the app. (For example, modify the front page template.)
+
+1.  Perform a new build with a new version tag "v2":
+
+        gcloud container builds submit --tag=gcr.io/${PROJECT_ID}/demo:v2 .
+
+    Now you have two builds stored in your project, `demo:v1` and `demo:v2`.
+    In general it's good practice to set the image tag for each build to a
+    unique build number. This will let you identify and deploy any build,
+    making updates and rollbacks easy.
+
+1.  Set the deployment to use the new image:
+
+        kubectl set image deployment/demo demo=gcr.io/${PROJECT_ID}/demo:v2
+
+    This performs a rolling update of all the running pods.
+
+1.  You can roll back to the earlier build by calling `kubectl set image`
+    again, specifying the earlier build tag.
+
+        kubectl set image deployment/demo demo=gcr.io/${PROJECT_ID}/demo:v1
+
+**Note:** If a deployment gets stuck because an error in the image prevents
+it from starting successfuly, you can recover by undoing the rollout. See the
+[Kubernetes deployment documentation](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/)
+for more info.
+
+## Clean up
+
+After you've finished this tutorial, you can clean up the resources you created
```

Comment:
After you've finished this tutorial, clean up the resources you created on Google Cloud Platform so you won't be billed for them going forward. To clean up, either delete your Kubernetes Engine resources, or delete the entire project.